devops/ansible/roles/girder/library/girder.py

```diff
@@ -942,10 +950,15 @@ def update(self, _id, body, **kwargs):
         if _id in self.resources:
             current = self.resources[_id]
             # if body is a subset of current we don't actually need to update
-            if set(body.items()) <= set(current.items()):
-                return current
-            else:
-                return self.__apply(_id, self.client.put, body, **kwargs)
+            try:
+                if set(body.items()) <= set(
```

Comment:
Where does this `body` come from? Is it the request body? Just trying to make sure it won't also contain any unhashable types.